### ISSUE TITLE
amp-stream-gallery: Non-text content has insufficient color contrast

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31159]
## WCAG Level
Level AA

## Priority
High

## Type
Color

## Pages/screens/components affected
* [amp-stream-gallery](https://amp.dev/documentation/components/amp-stream-gallery/)

## Description
An important design consideration is to make sure there is enough contrast between the color of meaningful non-text elements (such as controls denoted purely through graphical elements, icons used as the only visible indicator for errors, charts/graphs, meaningful borders/outlines used to denote a selected item) and their background. Good color contrast ensures that all sighted users, including those who have a condition that affects their ability to perceive color, are able to see/understand meaningful non-text content.

In the [amp-stream-gallery custom arrows](https://nutritious-tide-hornet.glitch.me/amp-stream-gallery/custom-arrows.html) example, the previous/next arrows have very low contrast.

## User impact
Without sufficient contrast, content can be difficult to perceive and therefore difficult to identify. In extreme cases, content with low contrast may not be visible at all to people with color perception impairments.

## Required solution
Make sure that non-text content has a color contrast ratio of at least 3:1 against its background and the content that is adjacent to it.

Elements that are disabled or inactive, and elements that use the browser's default styles, are exempt.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance
Currently, the custom styling for the arrows consists of a white (`#FFF`) stroke for the foreground arrow shape, and a soft drop shadow - leading to a contrast ratio that is below the minimum 3:1 ratio.

![The stream gallery with the low contrast previous/next buttons](https://user-images.githubusercontent.com/54933393/99181257-33b97f00-2725-11eb-9fdb-9ea94b4ca708.png)

```
.custom-arrow > svg {
  ...
  stroke: white;
  filter: drop-shadow(0 0 2px black);
}
```

There are many ways in which this styling can be modified to achieve a better contrast ratio. One possibility is to change the drop shadow to an actual outline/outside stroke.

![The stream gallery with the modified styling for the previous/next buttons](https://user-images.githubusercontent.com/54933393/99181268-48961280-2725-11eb-8bcc-875a132c4be2.png)

```
.custom-arrow > svg {
  ...
  filter: drop-shadow(0px 1px 0px #888) drop-shadow(0px -1px 0px #888) drop-shadow(1px 0px 0px #888) drop-shadow(-1px 0px 0px #888);
}
```

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Check the contrast of all non-text content using a tool such as the [TPG: Colour Contrast Analyser](https://developer.paciellogroup.com/resources/contrastanalyser/).
2. Check that the contrast level between the element and its adjacent content is at least 3:1.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website  has been tested for the same issue.
* All issues identified throughout the rest of the website  have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 1.4.11 Non-text Contrast (Level AA)](https://www.w3.org/TR/WCAG21/#non-text-contrast)

## More information
* [TPG: Colour Contrast Analyser](https://developer.paciellogroup.com/resources/contrastanalyser/)
* [WebAIM: Contrast and Color Accessibility - 1.4.11 Non-text Contrast](https://webaim.org/articles/contrast/#sc1411)
* [WebAIM: Color blindness](https://webaim.org/articles/visual/colorblind)

## Test data
Test date: November 2020
